examples/inject/src/main/java/io/helidon/examples/inject/ExternalContractExample.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.examples.inject;
+
+import java.util.List;
+import java.util.Random;
+
+import io.helidon.service.registry.Service;
+import io.helidon.service.registry.ServiceRegistryManager;
+
+/**
+ * An example that illustrates usages of {@link Service.ExternalContracts}.
+ */
+class ExternalContractExample {
+
+    private ExternalContractExample() {
+    }
+
+    /**
+     * A service that implements {@link CharSequence} contract.
+     */
+    @Service.PerLookup
+    @Service.ExternalContracts(CharSequence.class)
```

Comment:
`CharSequence` is an interface and as such will be an automatic contract. External contract is needed for classes only (i.e. an abstract class)

Reply:
Roger, used an abstract class instead.